pkgs/kde/gear/akonadi/default.nix

```diff
@@ -16,24 +26,40 @@ mkKdeDerivation {
     ./ignore-mysql-config-timestamp.patch
   ];
 
-  extraCmakeFlags = [
-    "-DMYSQLD_SCRIPTS_PATH=${lib.getBin mariadb}/bin"
-  ];
+  extraCmakeFlags =
+    [ "-DDATABASE_BACKEND=${lib.toUpper backend}" ]
+    ++ lib.optionals (backend == "mysql") [
+      "-DMYSQLD_SCRIPTS_PATH=${lib.getBin mariadb}/bin"
+    ]
+    ++ lib.optionals (backend == "postgres") [
+      "-DPOSTGRES_PATH=${lib.getBin postgresql}/bin"
+    ];
 
   extraNativeBuildInputs = [
     qttools
     shared-mime-info
   ];
-  extraBuildInputs = [
-    kaccounts-integration
-    accounts-qt
-    xz
-    mariadb
-  ];
+
+  extraBuildInputs =
+    [
+      kaccounts-integration
+      accounts-qt
+      xz
+    ]
+    ++ lib.optionals (backend == "mysql") [ mariadb ]
+    ++ lib.optionals (backend == "postgres") [ postgresql ]
+    ++ lib.optionals (backend == "sqlite") [ sqlite ];
 
   # Hardcoded as a QString, which is UTF-16 so Nix can't pick it up automatically
-  postFixup = ''
-    mkdir -p $out/nix-support
-    echo "${mariadb}" > $out/nix-support/depends
-  '';
+
+  postFixup =
+    lib.optional (backend == "mysql" || backend == "postgres") ''
```

Comment:
Can really be unconditional, an empty directory is whatever IMO instead of this readability disaster.

Reply:
it should now just create an empty folder for sqlite backend, for the rest if will behave like before.